### PR TITLE
Introduce CarService data service

### PR DESCRIPTION
**Pull request by manoldonev**
http://teampulse.telerik.com/view#item/346291
http://teampulse.telerik.com/view#item/346412

I stumbled upon an ObservableProperty decorator implementation [here](https://stackoverflow.com/questions/43311582/nativescript-class-implementing-observables-default-setter-behaviour) and I like it so I implemented it in the Car model class -- let me know what do you think (it probably should be extracted as a separate PR afterwards).

### CODE REVIEW

**Comment by manoldonev** [comment]
@MIYankov @ADjenkov removed the ObservableProperty decorator and replaced it with Observable.set(...) call where property change notification is necessary.

**Comment by manoldonev** [comment]
Added `"rxjs": "~5.4.0"` dependency to package.json

File: cars/shared/car-service.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    "transmission",
+    "class"
+];
+
+export class CarService {
+    static getInstance(): CarService {
+        return CarService._instance;
+    }
+
+    private static _instance: CarService = new CarService();
```
Comment (MIYankov) [inline cars/shared/car-service.ts]: Should this be static? Isn't it a property? Also, the properties should be on top. The _cars one too.

Comment (MIYankov) [inline cars/shared/car-service.ts]: Maybe properties -> constructor -> methods? for better readability?

File: cars/shared/car-service.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    private static cloneUpdateModel(car: Car) {
```
Comment (MIYankov) [inline cars/shared/car-service.ts]: Doesn't have a return type. Many others here doesn't have too.

File: cars/shared/car-service.ts (hunk 1 above)
Comment (manoldonev) [inline cars/shared/car-service.ts]: Current order is the one suggested by tslint.

File: cars/shared/car-service.ts (hunk 2 above)
Comment (manoldonev) [inline cars/shared/car-service.ts]: The return type here is `object`. Should we make it explicit?

Comment (manoldonev) [inline cars/shared/car-service.ts]: Ok, it seems `object` is more specific than `any` so I will update it (https://stackoverflow.com/questions/18961203/typescript-any-vs-object).

**Comment by manoldonev** [comment]
@MIYankov updated return types in CarService class.

**Pull request merged**